charts/sck-otel/templates/_config-Agent.tpl

```diff
@@ -359,13 +455,78 @@ service:
         - memory_limiter
         - batch
       exporters:
-        {{- if .Values.splunkPlatform.endpoint }}
+        {{- if eq (include "splunk-otel-collector.sendLogsToSplunk" .) "true" }}
         - splunk_hec/platform
         {{- end }}
-        {{- if .Values.splunkObservability.logsEnabled }}
-        {{- if or .Values.splunkObservability.ingestUrl .Values.splunkObservability.realm }}
+        {{- if eq (include "splunk-otel-collector.sendLogsToO11y" .) "true" }}
         - splunk_hec/o11y
         {{- end }}
+    {{- end }}
+    {{- if eq (include "splunk-otel-collector.collectTrace" .) "true" }}
+    # Default traces pipeline.
+    traces:
+      receivers: [otlp, jaeger, smartagent/signalfx-forwarder, zipkin]
+      processors:
+        - memory_limiter
+        - k8s_tagger
+        - batch
+        - resource/telemetry
+        - resourcedetection
+        {{- if .Values.environment }}
+        - resource/add_environment
+        {{- end }}
+      exporters:
+        {{- if .Values.gateway.enabled }}
+        - otlp
+        {{- else }}
+        - sapm
+        {{- end }}
+        {{- if eq (include "splunk-otel-collector.sendMetricsToO11y" .) "true" }}
+        # For trace/metric correlation.
+        - signalfx
+        {{- end }}
+    {{- end }}
+
+    {{- if eq (include "splunk-otel-collector.collectMetric" .) "true" }}
+    # Default metrics pipeline.
+    metrics:
+      receivers: [hostmetrics, kubeletstats, receiver_creator, signalfx]
+      processors:
+        - memory_limiter
+        - batch
+        - resource/telemetry
+        - resourcedetection
+      exporters:
+        {{- if .Values.gateway.enabled }}
+        - otlp
+        {{- else }}
+        {{- if eq (include "splunk-otel-collector.sendMetricsToO11y" .) "true" }}
+        - signalfx
+        {{- end }}
+        {{- if eq (include "splunk-otel-collector.sendMetricsToSplunk" .) "true" }}
+        - splunk_hec/platformMetrics
+        {{- end }}
+        {{- end }}
+    {{- end }}
+
+    {{- if eq (include "splunk-otel-collector.collectMetric" .) "true" }}
```

Comment:
These internal otel-collector metrics should be always sent to o11y if it's enabled. no matter which telemetry is enabled. so the condition here should be `splunk-otel-collector.sendMetricsToSplunk or splunk-otel-collector.splunkO11yEnabled`.